azkaban-common/src/main/java/azkaban/container/models/AzKubernetesV1PodBuilder.java

```diff
@@ -0,0 +1,189 @@
+package azkaban.container.models;
+
+import io.kubernetes.client.custom.Quantity;
+import io.kubernetes.client.openapi.models.V1Container;
+import io.kubernetes.client.openapi.models.V1ContainerBuilder;
+import io.kubernetes.client.openapi.models.V1EnvVar;
+import io.kubernetes.client.openapi.models.V1EnvVarBuilder;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import io.kubernetes.client.openapi.models.V1ObjectMetaBuilder;
+import io.kubernetes.client.openapi.models.V1Pod;
+import io.kubernetes.client.openapi.models.V1PodBuilder;
+import io.kubernetes.client.openapi.models.V1ResourceRequirements;
+import io.kubernetes.client.openapi.models.V1ResourceRequirementsBuilder;
+import io.kubernetes.client.openapi.models.V1Volume;
+import io.kubernetes.client.openapi.models.V1VolumeBuilder;
+import io.kubernetes.client.openapi.models.V1VolumeMount;
+import io.kubernetes.client.openapi.models.V1VolumeMountBuilder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * A wrapper Kubernetes Pod builder which internally utilizes
+ * @see io.kubernetes.client.openapi.models to create kubernetes "Pod" resource
+ *
+ * This class provides custom builder methods to have tight coupling of
+ * volumes and init containers (associated with job-types) with application-container
+ * enabling better type safety and compact implementation
+ *
+ * The Pod-spec created using this builder creates one application-container
+ * and multiple init containers corresponding to each jobType
+ */
+public class AzKubernetesV1PodBuilder {
```

Comment:
Sharing a thought on renaming it to AzKubernetesContainerBuilder? That way, it won't be dependent on kind of container we will be requesting.

Reply:
The problem is with the usage of the word container. K8s itself has both Pod and container terminology and it will be confusing if we write Container and mean Pod. Since we already have word Kubernetes in it, it should be ok. That way, the caller will appropriately invoke the builder class, in the current case, only this class.